File: cpp-starter/problems/sum.cpp
#include <string>
#include <fstream>
#include <iostream>

#include "sum.hpp"

// Initialize the parser for the sum problem
void SumProblem::initialize_parser(cxxopts::Options &options) {
    options.add_options()
        ("sum", "Sum the numbers in the input file");
}

// Check if the sum problem is chosen
bool SumProblem::is_chosen_problem(const cxxopts::ParseResult &args) {
    return args.count("sum") > 0;
}

// Run the sum problem
int SumProblem::run(const cxxopts::ParseResult &args) {
    std::string inputFilename = args["input"].as<std::string>();
    std::string outputFilename = args["output"].as<std::string>();

    std::ifstream inputFile(inputFilename);

    if (!inputFile) {
        std::cerr << "Error opening input file: " << inputFilename << std::endl;
        return 1;
    }

    int sum = 0;
    int number;

    while (inputFile >> number) {
        sum += number;
    }

    std::ofstream outputFile(outputFilename);

    if (!outputFile) {
        std::cerr << "Error opening output file: " << outputFilename << std::endl;
        return 1;
    }

    outputFile << sum;
    return 0;
}

File: python-starter/project/problems/sum.py
from project.problem import Problem
import argparse

class SumProblem(Problem):

    def initialize_parser(self, parser: argparse.ArgumentParser):
        """
        Initialize the parser with the necessary arguments
        """
        parser.add_argument('--sum', help='sum the given numbers', action='store_true')
    
    def is_chosen_problem(self, args):
        """
        Check if the problem is chosen
        """
        return bool(args.sum)

    def run(self, args):
        """
        Run the program
        """
        # Access the input and output file paths
        input_file = args.input
        output_file = args.output

        # Read the numbers
        with open(input_file, 'r') as f:
            data = f.read()
            
        numbers = data.split(',')
        numbers = [int(number) for number in numbers]

        # Process the numbers
        result = sum(numbers)

        # Write the result into the file
        with open(output_file, 'w') as f:
            f.write(str(result))
